Java/export-api-sample/src/main/java/com/perfecto/reporting/sample/api/ReportiumExportUtils.java

```diff
@@ -197,21 +241,63 @@ private static void downloadTestReport(Path testPdfPath, CreatePdfTask task, Str
         while (updatedTask.getStatus() != TaskStatus.COMPLETE && startTime + maxGenerationTime > System.currentTimeMillis());
 
         if (updatedTask.getStatus() == TaskStatus.COMPLETE) {
-            downloadPdfFileToFS(testPdfPath, new URI(updatedTask.getUrl()));
+            downloadPdfOrCsvFileToFS(testPdfPath, new URI(updatedTask.getUrl()));
         } else {
             throw new RuntimeException("The task is still in " + updatedTask.getStatus() + " status after waiting " + maxWaitMin + " min");
         }
     }
 
-    private static CreatePdfTask getUpdatedTask(String taskId) throws URISyntaxException, IOException {
-        CreatePdfTask task;
+    private static void downloadCsvTestReport(Path testCsvPath, CreateTask task) throws URISyntaxException, IOException {
+        System.out.println("Downloading CSV");
+        long startTime = System.currentTimeMillis();
+        int maxWaitMin = 10;
+        long maxGenerationTime = TimeUnit.MINUTES.toMillis(maxWaitMin);
+        String taskId = task.getTaskId();
+
+        CreateTask updatedTask;
+        do {
+            updatedTask = getUpdatedCsvTask(taskId);
+            try {
+                if (updatedTask.getStatus() != TaskStatus.COMPLETE) {
+                    Thread.sleep(3000);
+                }
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        while (updatedTask.getStatus() != TaskStatus.COMPLETE && startTime + maxGenerationTime > System.currentTimeMillis());
+
+        if (updatedTask.getStatus() == TaskStatus.COMPLETE) {
+            downloadPdfOrCsvFileToFS(testCsvPath, new URI(updatedTask.getUrl()));
+        } else {
+            throw new RuntimeException("The task is still in " + updatedTask.getStatus() + " status after waiting " + maxWaitMin + " min");
+        }
+    }
+
+    private static CreateTask getUpdatedTask(String taskId) throws URISyntaxException, IOException {
```

Comment:
this method and the following method that you added do basically the same thing. The only difference is that the URL is different. This means that you can create 1 method that receives the URL as a param. Also, create global constants for `REPORTING_SERVER_URL + "/export/api/v2/test-executions/pdf/task/"` and `REPORTING_SERVER_URL + "/export/api/v2/test-executions/csv/"`

Reply:
updated the code